includes/class-sensei-learner.php

```diff
@@ -99,6 +102,174 @@ public function delete_all_user_activity( $user_id = 0 ) {
 		return $dataset_changes;
 	}
 
+	/**
+	 * Query the courses a user is enrolled in.
+	 *
+	 * @param int   $user_id         User ID.
+	 * @param array $base_query_args Base query arguments.
+	 *
+	 * @return WP_Query
+	 */
+	public function get_enrolled_courses_query( $user_id, $base_query_args = [] ) {
+		/**
+		 * Fire before we query a user's enrolled courses.
+		 *
+		 * @since 3.0.0
+		 *
+		 * @param int $user_id User ID.
+		 */
+		do_action( 'sensei_before_learners_enrolled_courses_query', $user_id );
+
+		$query_args = $this->get_enrolled_courses_query_args( $user_id, $base_query_args );
+
+		return new WP_Query( $query_args );
+	}
+
+	/**
+	 * Query the courses a user is enrolled in and hasn't completed.
+	 *
+	 * @param int   $user_id         User ID.
+	 * @param array $base_query_args Base query arguments.
+	 *
+	 * @return WP_Query
+	 */
+	public function get_enrolled_active_courses_query( $user_id, $base_query_args = [] ) {
+		$base_query_args[ self::COURSE_PROGRESS_FILTER ] = self::COURSE_PROGRESS_FILTER_ACTIVE;
+
+		return $this->get_enrolled_courses_query( $user_id, $base_query_args );
+	}
+
+	/**
+	 * Query the courses a user is enrolled in and has completed.
+	 *
+	 * @param int   $user_id         User ID.
+	 * @param array $base_query_args Base query arguments.
+	 *
+	 * @return WP_Query
+	 */
+	public function get_enrolled_completed_courses_query( $user_id, $base_query_args = [] ) {
+		$base_query_args[ self::COURSE_PROGRESS_FILTER ] = self::COURSE_PROGRESS_FILTER_COMPLETED;
+
+		return $this->get_enrolled_courses_query( $user_id, $base_query_args );
+	}
+
+	/**
+	 * Get the arguments to pass to WP_Query to fetch a learner's enrolled courses.
+	 *
+	 * @param int   $user_id         User ID.
+	 * @param array $base_query_args Base query arguments.
+	 *
+	 * @return array
+	 */
+	public function get_enrolled_courses_query_args( $user_id, $base_query_args = [] ) {
+		$order                = 'DESC';
+		$orderby              = 'date';
+		$has_set_course_order = '' !== get_option( 'sensei_course_order', '' );
+
+		// If a fixed course order has been set, trust menu_order.
+		if ( $has_set_course_order ) {
+			$order   = 'ASC';
+			$orderby = 'menu_order';
+		}
+
+		$default_args = [
+			'post_status' => 'publish',
+			'order'       => $order,
+			'orderby'     => $orderby,
+			'tax_query'   => [], // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_tax_query -- Just empty to set array.
+		];
+
+		$query_args   = array_merge( $default_args, $base_query_args );
+		$learner_term = self::get_learner_term( $user_id );
+
+		$query_args['post_type']   = 'course';
+		$query_args['tax_query'][] = [
+			'taxonomy'         => Sensei_PostTypes::LEARNER_TAXONOMY_NAME,
+			'terms'            => $learner_term->term_id,
+			'include_children' => false,
+		];
+
+		if ( isset( $query_args[ self::COURSE_PROGRESS_FILTER ] ) ) {
```

Comment:
I'm doing this here instead of the individual methods above because `sensei_before_learners_enrolled_courses_query` needs to run beforehand. Alternatively, we could put that `do_action` in a separate method and these in their respective helper method above.